### PR TITLE
adb reboot bootloader/fastboot/recovery work well

### PR DESCRIPTION
**Pull request by xiaopingx**
For now, we can use
       adb reboot bootloader
       adb reboot-bootloader
       adb reboot fastboot
    boot into fastboot mode

       adb reboot recovery
    boot into recovery mode

Bug ID: https://01.org/jira/browse/AIA-159
Test: Test on Joule board, them could work well

Signed-off-by: hu,xiaopingx <xiaopingx.hu@intel.com>
Signed-off-by: Chen,Xihua <xihua.chen@intel.com>

### CODE REVIEW

**Comment by xiaopingx** [comment]
https://github.com/android-ia/device-intel-common/pull/2
https://github.com/android-ia/system-core/pull/3
https://github.com/android-ia/bootable-recovery/pull/2
https://github.com/android-ia/device-androidia-mixins/pull/31
https://github.com/android-ia/device-androidia/pull/39

File: init/builtins.cpp [hunk 1]
```diff
@@ -704,16 +705,27 @@ static int do_powerctl(const std::vector<std::string>& args) {
             callback_on_ro_remount = unmount_and_fsck;
         } else if (cmd == ANDROID_RB_RESTART2) {
             reboot_target = &command[len + 1];
+#ifdef BUILD_JOULE_BOOTLOADER_CONTROL
```
Comment (shobhitka) [inline init/builtins.cpp]: Are the following changes specific to Joule ? Looks like they should be generically there without this switch and needed on all targets ?

File: init/Android.mk [hunk 2]
```diff
@@ -111,6 +113,11 @@ LOCAL_POST_INSTALL_CMD := $(hide) mkdir -p $(TARGET_ROOT_OUT)/sbin; \
 
 LOCAL_SANITIZE := integer
 LOCAL_CLANG := true
+
+ifeq ($(BUILD_JOULE_BOOTLOADER_CONTROL),true)
+        LOCAL_CFLAGS = -DBUILD_JOULE_BOOTLOADER_CONTROL
+endif
```
Comment (shobhitka) [inline init/Android.mk]: Why we need to have this flag in the first place. Code looks generic enough to be enabled

**Comment by shobhitka** [comment]
You might also have to rebase this patch on the latest code and resubmit.

**Pull request closed without merging**